### PR TITLE
Edit: Refactor away from `MessageProvider` and `Recipe`

### PR DESCRIPTION
**Pull request by umpox**
## Description

Edit is now:
- Using it's own logic instead of sharing `MessageProvider`. 
- No longer a recipe, all logic lives inside of `/edit` and `non-stop`. In a follow up PR I will consolidate the logic in `non-stop` into this directory too and clean that up.
- Refactored to easily support changes without affecting other commands/chat/recipes.
- Refactored to support easily adding prompts for new models such as GPT4

closes https://github.com/sourcegraph/cody/issues/2545

## Test plan

Followed edit test plans in https://docs.google.com/document/d/1all3z8dPs52amhNs3m4O1Lu7V6VsCaU7jXovprte-EE/edit#heading=h.o8dycc3mlio

Followed telemetry test plan via https://github.com/sourcegraph/cody/issues/2545

### CODE REVIEW

**Review by abeatrix — CHANGES_REQUESTED** [review]
I tried the build from this branch but the comments are not working for me:


Doc command is not working:

https://github.com/sourcegraph/cody/assets/68532117/7ba40757-2dfd-4ae2-a419-4727adb28a9d

Edit command is also not working:

https://github.com/sourcegraph/cody/assets/68532117/8dea235c-fd6b-4cdc-99ff-219c81c4d18d

Other non-edit commands will open a new chat panel but they are all stuck in loading screen: 

image

**Comment by umpox** [comment]
@abeatrix I can't replicate 🤔 Do you have any logs in your console (VS Code dev tools or the Cody Output log)? The diff should definitely not affect any chat related code, it seems like the extension hasn't activated properly.

https://github.com/sourcegraph/cody/assets/9516420/ac80ef28-e317-4269-917c-af5e10fa927a

**Review by abeatrix — APPROVED** [review]
That was my bad. I was testing it on files that were added to my `.cody/ignore` file so they were not showing up as expected 😅 

Confirmed the edit commands are working correctly after the change!

Uploading Screen Recording 2024-01-08 at 9.26.02 AM.mov…

File: vscode/src/edit/manager.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        telemetryService.log(
+            `CodyVSCodeExtension:command:${commandEventName}:executed`,
+            { source },
+            { hasV2Event: true }
+        )
+        telemetryRecorder.recordEvent(`cody.command.${commandEventName}`, 'executed', { privateMetadata: { source } })
+
+        const editor = getEditor()
+        if (editor.ignored) {
+            console.error('File was ignored by Cody.')
```
Comment (abeatrix) [inline vscode/src/edit/manager.ts:60]:
```suggestion
            void vscode.window.showInformationMessage('Cannot edit Cody ignored file.')
```

**Comment by abeatrix** [comment]
we can also remove 'registerTreeView' and `taskViewProvider` from FixupController?

**Comment by umpox** [comment]
@abeatrix I think so, I'll do it in a follow up PR

File: vscode/src/edit/manager.ts (hunk 1 above)
Comment (umpox) [reply vscode/src/edit/manager.ts]: Good idea 👍

**Pull request merged**